tc/lang/sema.h

```diff
@@ -339,6 +339,21 @@ struct Sema {
         throw ErrorReport(exp) << "NYI - semantic checking for " << exp;
     }
   }
+  // This is the entry function for semantic analysis. It is called by
+  // tc2halide to associate type with each node of the tree and to also make
+  // sure that the tree is sematically correct. For example: a variable
+  // may not be input two times. Parser only verifies for the syntax but does
+  // not check the semantics.
+  //
+  // It converts the TK_APPLY nodes to TK_ACCESS or TK_BUILT_IN
+  //
+  // The reduction variables are deduced and the objects are created for them
+  // and they are appended to the tree
+  //
+  // Type checking is also done but by a very small codebase
+  //
+  // withType - associates the type with a given node
```

Comment:
What's the connection with `checkFunction`?
That is, why did you add this here?

Reply:
checkFunction is the main entry point for Sema codebase called by `tc2halide`

I added `withType` because this is important component of sema where each node gets it's proper type. I wanted to highlight that it's an important functionality as part of what `checkFunction` will result it.